src/vs/platform/terminal/common/terminal.ts

```diff
@@ -167,14 +167,20 @@ export interface IPtyHostAttachTarget {
 	icon: TerminalIcon | undefined;
 	fixedDimensions: IFixedTerminalDimensions | undefined;
 	environmentVariableCollections: ISerializableEnvironmentVariableCollections | undefined;
-	reconnectionOwner?: string;
-	task?: { label: string; id: string; lastTask: string; group?: string };
+	reconnectionProperties?: IReconnectionProperties;
 	waitOnExit?: WaitOnExitValue;
 	hideFromUser?: boolean;
 	isFeatureTerminal?: boolean;
 	type?: TerminalType;
 }
 
+export interface IReconnectionProperties {
+	ownerId: string;
+	args?: IReconnectionTaskArgs;
+}
+
+export interface IReconnectionTaskArgs { label: string; id: string; lastTask: string; group?: string }
```

Comment:
my thought is it would be confusing to call this something generic. as we have other owners, we can just set `args` type to be `IReconnectionTaskArgs` | `IReconnectionOtherOwnerArgs`

Reply:
`args` isn't the best name here as it's not provided to a function, how about `data`?

Also could we not have this as `data?: unknown` instead and then let the embedder validate/cast to avoid talking about tasks in terminal? Since tasks is the only thing saving/fetching it, the cast should always work.